wave_lang/kernel/wave/analysis/partition_strided_operators.py

```diff
@@ -380,3 +387,117 @@ def has_gpr_offsets(node: fx.Node) -> bool:
+
+    The pass:
+    1. Identifies reads/writes with non-contiguous access patterns
+    2. For each such operation:
+       - Creates multiple single-element reads/writes
+       - Updates indices to access the correct elements
+       - Handles dynamic values in mappings
+       - Combines results back together if needed (for reads)
+    """
+
+    def has_gather_mapping(node: fx.Node) -> bool:
+        """
+        Checks for writes on 2d tensors with strided access on a single dimension that
+        read more than a single element.
+        """
+        custom = get_custom(node)
+        if isinstance(custom, (Read, Write)):
+            return not custom.is_contiguous_vec()
+
+        return False
+
+    strided_operators = trace.walk(has_gather_mapping)
+    for operator in strided_operators:
+        custom = get_custom(operator)
+        index = custom.index
+        fastest_index = get_fastest_index(index)
+        elements_per_thread = subs_idxc(custom.elements_per_thread)
+
+        # Break apart Reads/Writes that has non-contiguous GPR Read/Writes.
+        with custom.graph.inserting_before(operator):
+            ops_to_combine = []
+            for i in range(elements_per_thread):
+                new_index = deepcopy(index)
+                for j, v in enumerate(new_index.values()):
+                    if j == fastest_index:
+                        v.start = v.start + i
+                    v.size = 1
+                    v.stride = 1
+
+                new_dynamic_vals = []
+                for dynamic_val in custom.mapping_dynamic_vals:
+                    _, size = get_largest_index_and_size(dynamic_val.index)
+                    if size == 1:
+                        new_dynamic_vals.append(dynamic_val)
+                        continue
+
+                    assert (
+                        size == elements_per_thread
+                    ), f"Expected size to be equal to {elements_per_thread}, got {size}"
+
+                    extract = ExtractSlice(dynamic_val, [i], [1], [1]).add_to_graph(
+                        custom.graph
+                    )
+                    extract.index = new_index
+                    new_dynamic_vals.append(extract)
+
+                # Generate new Read/Write that has contiguous VGPR elements.
+                if isinstance(custom, Write):
+                    extract = ExtractSlice(
+                        custom.register_, [i], [1], [1]
+                    ).add_to_graph(custom.graph)
+                    extract.index = new_index
+                    new_node = Write(
+                        extract,
+                        custom.memory,
+                        mapping=custom.mapping,
+                        mapping_dynamic_vals=new_dynamic_vals,
+                        elements_per_thread=1,
+                    ).add_to_graph(custom.graph)
+                elif isinstance(custom, Read):
+                    # TODO: Add support on how to handle strided reads.
```

Comment:
Why won't the existing code handle strided reads? And what about strides writes?